README.md

```diff
@@ -1 +1,47 @@
-# pfcpsim
\ No newline at end of file
+# pfcpsim
+pfcpsim is a simulator to interact with PFCP agents. Can be used to emulate a 4G SGW / 5G UPF.
+
+## Interaction
+pfcpsim is designed to work within a containerized environment. The docker image comes with both client (`pcpfsimctl`) and server (`pfcpsim`).
+
+`PFCPClient` is embedded in a gRPC Server. Interaction between pfcpsim and pfcpsimctl is performed through RPC calls, as shown in the following schema: 
+
+![Alt text](docs/images/schema.svg?raw=true "schema")
+
+## Example
+In the following example we will borrow `mock-up4` environment, from [upf-epc repository](https://github.com/omec-project/upf-epc/tree/master/test/integration).
+Using mock-up4's `docker-compose.yml` will create a pfcp agent container listening on `0.0.0.0:8805`:
+
+1. Build the container locally (You can skip this once docker image is published by the CI):
+```bash
+make build-pfcpsim
+```
+2. Create the container using host network (to be able to connect to 127.0.0.1:8805)
+```bash
+docker container run --rm -d --name pfcpsim --net host pfcpsim:0.1.0-dev
+```
+
+3. Use pfcpsimctl by using `docker exec` on the same container to configure the server (not passing any flag will use mock-up4's default values)
```

Comment:
```suggestion
3. Use `pfcpctl` to configure the server (not passing any flag will use mock-up4's default values)
```

Reply:
what are the mock-up4's default values? We shouldn't use names (mock-up4) specific to other projects (upf-epc in this case).